Feat188 patterns weaves (#192)
federated-research/docs: website/pages/federated_research_patterns/egress.mdx

> # Data Egress Patterns

Reviewer: I wonder whether it would it be helpful to add further context about this being relevant for the Safe Outputs. 

I'm thinking for example, this guidance: https://www.gov.uk/government/publications/safeguarding-uk-human-genomic-data/safeguarding-uk-human-genomic-data

I find it interesting that the Safe Outputs section there covers similar ground briefly, but using very different language, and they never uses the word Egress either.

Author: Yeah, agree. I was wanting to wait for the wider docs to see what we would have on five safes - there is the current pages and was hoping they would be updated.

> There are a few mechanisms in which egress can occur at the boundary of the TRE and/or at the boundary of the federated network. 

Reviewer: I wonder as you're referring to these the terms local and central in each pattern, it could be helpful for readers to use those words in this opening explanation?

> ## Disclosure Control Components

Reviewer: Should this be Egress Control Components? 

Because it looks like when used as a component (lines 13, 19, 25, 31) it is referred to as that. 

I'm just a little confused when I read as a page, as I am introduced to `Disclosure Control Components` but then referred to as something else when being implemented.

- **None** – where the egress being requested is essentially set to auto-approve, there can be no manual or automated egress check in place, or no process to mitigate disclosure risks. 
- **Manual** - where at least one person checks the results by eye and approves the release of the data

> - **Semi-automated** – where at least one person checks the results, aided by a risk tool

Reviewer 2: I think we need to say a bit more about what we mean by a risk tool. 
I think we mean something like the automated case, but with a human decision. I want to be clear as to what level of involvement the software tools have in semi automated vs automated vs manual. 
I think ACRO falls very much in semi-automated - it does some checks, some processing and produces a report to support the decision being made by a human. If someone takes data, puts it into a spreadsheet and does some risk calculations themselves, or using excel formulae, I feel like that's manual. Is it still manual if the spreadsheet calculates some sort of risk scores automatically, or using a macro? or is that semi-auto? Where is the line?

- **Automated** – where a computational approach has been taken to check contents or mitigate disclosure risks, and make the decision with no human involvement. 

## Current Patterns

### Full local control

> Egress-control local: Manual <br/>

Reviewer: I notice you're using html in markdown here - generally we would lean away from doing this unless there's a benefit, in this case just more lines in the markdown would be better

Author: Yeah - help a markdown novice, what should it be? I picked this up I think from somewhere else in the doc and was surprised.

Author: Double space? That is horrible.

Author: fixing this

Egress-control central: Any  <br/>

> This egress patterns is most likely affiliated to the Traditional Federation pattern as it mirrors current process as much as possible.All data to move out from the boundary of the individual TRE has a manual disclosure control check, regardless if any disclosure control is preformed on the federated network.

Reviewer 3: ```suggestion
This egress pattern is most likely affiliated with the Traditional Federation pattern, as it mirrors the current process as much as possible. All data to move out from the boundary of the individual TRE has a manual disclosure control check, regardless of whether any disclosure control is performed on the federated network.
```

> This egress pattern is most likely affiliated with the Traditional Federation pattern, as it mirrors the current process as much as possible. All data to move out from the boundary of the individual TRE has a manual disclosure control check, regardless of whether any disclosure control is performed on the federated network.

Reviewer: Do you think it would be worthwhile to deep link to the Traditional Federation pattern when you refer to it here? I find myself having to go back to it and uncovering it

### Semi-automated
Egress-control local: Semi-automated <br/>
Egress-control central: Any  <br/>

> There are emerging tools (such as [SACRO](https://github.com/AI-SDC)) that can allow for the semi-automated checking of results for a disclosure risk. The decision to release a result is still manual and undertaken by the TRE, with the aim this semi-automation can first warn the researcher ahead of time that a result may be disclosive and therefore accelerate any disclosure control assessments undertake at the time of egress from the TRE. 

Reviewer 3: ```suggestion
There are emerging tools (such as [SACRO](https://github.com/AI-SDC)) that can allow for the semi-automated checking of results for a disclosure risk. The decision to release a result is still manual and undertaken by the TRE, with the aim that this semi-automation can first warn the researcher ahead of time that a result may be disclosive and therefore accelerate any disclosure control assessments undertaken at the time of egress from the TRE. 
```

### Fully automated
Egress-control local: Automated <br/>
Egress-control central: Any  <br/>

> There are federated platforms and tools that will automate the checks and assure that data which is moving is not disclosive before it leaves the boundary of the TRE. These checks are fully automated and therefore there is no human in the loop to check the results, instead, the software/algorithm is valdiated to give the assurance to the TRE that the correct checks and balances are applied to the data before it moves. 

Reviewer 3: ```suggestion
There are federated platforms and tools that will automate the checks and ensure that data which is moving is not disclosive before it leaves the boundary of the TRE. These checks are fully automated, and therefore, there is no human in the loop to check the results; instead, the software/algorithm is validated to give assurance to the TRE that the correct checks and balances are applied to the data before it moves. 
```

### Full central control
Egress-control local: None 

> Egress-control central: Any 

Reviewer: ```suggestion
Egress-control local: None 

Egress-control central: Any 
```

You haven't used a linebreak here so they are on the sameline

> There are certain analyses that can only be performed when all the data is in a single location. Therefore data pooling must occur and data that normally would not be allowed via egress does indeed move. The key concept of this pattern though is that the data is moving to another TRE/SDE and that before the results leaves the boundary of the federation of TREs, the normal egress check is performed. At this stage it could be any of the disclosure control components that are used, but would be agreed by the federation of TREs/SDEs. 

Reviewer 3: ```suggestion
There are certain analyses that can only be performed when all the data is in a single location. Therefore, data pooling must occur, and data that normally would not be allowed via egress does indeed move. The key concept of this pattern, though, is that the data is moving to another TRE/SDE and that before the results leave the boundary of the federation of TREs, the normal egress check is performed. At this stage, it could be any of the disclosure control components that are used, but would be agreed upon by the federation of TREs/SDEs. 
```

> There are certain analyses that can only be performed when all the data is in a single location. Therefore, data pooling must occur, and data that normally would not be allowed via egress does indeed move. The key concept of this pattern, though, is that the data is moving to another TRE/SDE and that before the results leave the boundary of the federation of TREs, the normal egress check is performed. At this stage, it could be any of the disclosure control components that are used, but would be agreed upon by the federation of TREs/SDEs. 

Reviewer: ```suggestion
There are certain analyses that can only be performed when all the data is in a single location. Therefore, data pooling must occur, and data that normally would not be allowed via egress does indeed move. The key concept of this pattern, is that the data is moving to another TRE/SDE and that before the results leave the boundary of the federation of TREs, the normal egress check is performed. At this stage, it could be any of the disclosure control components that are used, but would be agreed upon by the federation of TREs/SDEs. 
```
`though` doesn't add much here.

> There are certain analyses that can only be performed when all the data is in a single location. Therefore, data pooling must occur, and data that normally would not be allowed via egress does indeed move. The key concept of this pattern, though, is that the data is moving to another TRE/SDE and that before the results leave the boundary of the federation of TREs, the normal egress check is performed. At this stage, it could be any of the disclosure control components that are used, but would be agreed upon by the federation of TREs/SDEs. 

Reviewer 2: This kind of egress pattern would also be commonly used in FL (so in connected analyses). Do you consider that data pooling (because it kind of is, but not in the same way that we refer to it in the patterns context)? Is it worth a mention here, or a rephrasing of 'data pooling' to avoid it seeming like you're referring to pooled pattern exclusively?

> # Data E-gress Patterns

Reviewer 4: ```suggestion
# Data Egress Patterns
```

> There are a few mechanisim in which e-gress can occur at the boundary of the TRE and/or at the boundary of the federated network. 

Reviewer 4: ```suggestion
There are a few mechanisms in which egress can occur at the boundary of the TRE and/or at the boundary of the federated network. 
```

## Disclosure Control Components

> On aspect that is likely to have a pattern are the various methods in which the federated platorms allow (or not) for disclosure control checking to occur at the level of the individual TRE. 

Reviewer 4: ```suggestion
One aspect that is likely to have a pattern are the various methods in which the federated platorms allow (or not) for disclosure control checking to occur at the level of the individual TRE. 
```

The grammar rubs me the wrong way here but i don't know how to fix it ("One aspect [...] are the various methods" reads weird) but caught the typo anyhow.

- **None** – where the egress being requested is essentially set to auto-approve, there can be no manual or automated egress check in place, or no process to mitigate disclosure risks. 
- **Manual** - where at least one person checks the results by eye and approves the release of the data
- **Semi-automated** – where at least one person checks the results, aided by a risk tool
- **Automated** – where a computational approach has been taken to check contents or mitigate disclosure risks, and make the decision with no human involvement. 

## Current Patterns

### Full local control
Egress-control local: Manual <br/>
Egress-control central: Any  <br/>

> This e-gress patterns is most likely affiliated to the Traditional Federation pattern as it mirrors current process as much as possible.All data to move out from the boundary of the individual TRE has a manual disclosure control check, regardless if any disclosure control is preformed on the federated network.

Reviewer 4: ```suggestion
This egress pattern is most likely affiliated to the Traditional Federation pattern as it mirrors current process as much as possible.All data to move out from the boundary of the individual TRE has a manual disclosure control check, regardless if any disclosure control is preformed on the federated network.
```

### Semi-automated
Egress-control local: Semi-automated <br/>
Egress-control central: Any  <br/>

> There are emerging tools (such as SACRO) that can allow for the semi-automated checking of results for a disclosure risk. The decision to release a result is still manual and undertaken by the TRE, with the aim this semi-automation can first warn the researcher ahead of time that a result may be disclosive and therefore accelerate any disclosure control assessments undertake at the time of egress from the TRE. 

Reviewer 4: > such as SACRO

worth a link here?
